etl/controller/pulsemaintainer.go

```diff
@@ -99,9 +102,81 @@ func (c *Controller) pulseSequence(ctx context.Context) {
 }
 
 func pulseIsComplete(p types.Pulse, d []string) bool { // nolint
-	// TODO implement me
-	// This if is here for test reason, delete it after implementation and update test data for expected behavior
-	return p.PulseNo >= 0
+	if len(d) == 0 {
+		return false
+	}
+
+	// root
+	if len(d) == 1 && d[0] == "" {
+		return true
+	}
+
+	// inverts last bool symbol in string
+	invertLastSymbol := func(s string) string {
+		last := s[len(s)-1:]
+		var invertedSymbol string
+		b, err := strconv.ParseBool(last)
```

Comment:
Why parse symbol? Why not  `if last == "0" ... else ...` ?

Reply:
nice, done